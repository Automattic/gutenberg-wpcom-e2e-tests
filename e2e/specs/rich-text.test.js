/**
 * Internal dependencies
 */
import {
	newPost,
	getEditedPostContent,
	insertBlock,
	clickBlockAppender,
	pressWithModifier,
	META_KEY,
	ACCESS_MODIFIER_KEYS,
} from '../support/utils';

describe( 'RichText', () => {
	beforeEach( async () => {
		await newPost();
	} );

	it( 'should handle change in tag name gracefully', async () => {
		// Regression test: The heading block changes the tag name of its
		// RichText element. Historically this has been prone to breakage,
		// specifically in destroying / reinitializing the TinyMCE instance.
		//
		// See: https://github.com/WordPress/gutenberg/issues/3091
		await insertBlock( 'Heading' );
		await page.click( '[aria-label="Heading 3"]' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should apply formatting with access shortcut', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await pressWithModifier( META_KEY, 'a' );
		await pressWithModifier( ACCESS_MODIFIER_KEYS, 'd' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should apply formatting with primary shortcut', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await pressWithModifier( META_KEY, 'a' );
		await pressWithModifier( META_KEY, 'b' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should apply formatting when selection is collapsed', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'Some ' );
		// All following characters should now be bold.
		await pressWithModifier( META_KEY, 'b' );
		await page.keyboard.type( 'bold' );
		// All following characters should no longer be bold.
		await pressWithModifier( META_KEY, 'b' );
		await page.keyboard.type( '.' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
