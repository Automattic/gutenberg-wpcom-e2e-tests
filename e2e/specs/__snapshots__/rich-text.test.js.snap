// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText should apply formatting when selection is collapsed 1`] = `
"<!-- wp:paragraph -->
<p>SomeÂ <strong>bold</strong>.</p>
<!-- /wp:paragraph -->"
`;

exports[`RichText should apply formatting with access shortcut 1`] = `
"<!-- wp:paragraph -->
<p><del>test</del></p>
<!-- /wp:paragraph -->"
`;

exports[`RichText should apply formatting with primary shortcut 1`] = `
"<!-- wp:paragraph -->
<p><strong>test</strong></p>
<!-- /wp:paragraph -->"
`;

exports[`RichText should handle change in tag name gracefully 1`] = `
"<!-- wp:heading {\\"level\\":3} -->
<h3></h3>
<!-- /wp:heading -->"
`;
